docs/proposals/failover/optimize-cluster-failover.md

```diff
@@ -0,0 +1,635 @@
+    // +optional
+    TargetCluster *ClusterAffinity `json:"targetCluster,omitempty"`
+
+    // MatchConditions indicates the conditions to match for triggering
+    // the taint controller to add taints on the cluster object.
+    // The match conditions are ANDed.
+    // If empty, the TaintsToAdd will be added.
+    // +optional
+    MatchConditions []MatchCondition `json:"matchConditions,omitempty"`
+
+    // TaintsToAdd specifies the NoExecute effect taints that
+    // need to be applied to the clusters which match with ClusterAffinity.
+    // +optional
+    TaintsToAdd []TaintToAdd `json:"taintsToAdd,omitempty"`
+}
+
+// MatchCondition represents the condition match detail of activating the failover relevant taints on clusters.
+type MatchCondition struct {
+    // ConditionType specifies the ClusterStatus condition type.
+    ConditionType string `json:"conditionType"`
+
+    // Operator represents a relationship to a set of values.
+    // Valid operators are In, NotIn.
+    Operator MatchConditionOperator `json:"operator"`
+
+    // StatusValues is an array of metav1.ConditionStatus values.
+    // The item specifies the ClusterStatus condition status.
+    StatusValues []metav1.ConditionStatus `json:"statusValues"`
+}
+
+// A MatchCondition operator is the set of operators that can be used in the match condition.
+type MatchConditionOperator string
+
+const (
+    MatchConditionOpIn    MatchConditionOperator = "In"
+    MatchConditionOpNotIn MatchConditionOperator = "NotIn"
+)
+
+// TaintToAdd describes the taint that needs to be applied to the cluster.
+type TaintToAdd struct {
+    // Key represents the taint key to be applied to a cluster.
+    // +required
+    Key string `json:"key"`
+
+    // Effect represents the taint effect to be applied to a cluster.
+    // +required
+    Effect  corev1.TaintEffect `json:"effect"`
+
+    // Value represents the taint value corresponding to the taint key.
+    // +optional
+    Value string `json:"value,omitempty"`
+
+    // AddOnMatchSeconds is the duration in seconds for which the
+    // MatchConditions must be continuously matched before adding the
+    // specified taint to the cluster.
+    // This provides stability by ensuring the condition persists long enough
+    // before taking action.
+    // Defaults to 300 seconds (5 minutes).
+    // +optional
+    // +kubebuilder:default=300
+    // +kubebuilder:validation:Minimum=1
+    AddOnMatchSeconds *int32 `json:"addOnMatchSeconds,omitempty"`
+
+    // RemoveOnMismatchSeconds is the duration in seconds for which the
+    // MatchConditions must be continuously unmatched before removing
+    // the taint from the cluster.
+    // This prevents premature taint removal during brief connection recoveries.
+    // Defaults to 180 seconds (3 minutes).
+    // +optional
+    // +kubebuilder:default=180
+    // +kubebuilder:validation:Minimum=1
+    RemoveOnMismatchSeconds *int32 `json:"removeOnMismatchSeconds,omitempty"`
```

Comment:
Leaving the `AddOnMatchSeconds` and `RemoveOnMismatchSeconds` in `TaintToAdd` means each taint would have a particular configuration. Is it right? What's the expected use case?

In my opinion, `AddOnMatchSeconds` and `RemoveOnMismatchSeconds` should be at the policy level, each `ClusterTaintPolicy` declares a match condition, and once matched, all taints should be added/removed simultaneously. The benefits include:
- simplify the configuration that don't have to replicate for each taint. (If a taint really need a particular period configuration, start another `ClusterTaintPolicy`)
- simplify the implementation logic